lib/api/v3/users/user_representer.rb

```diff
@@ -58,6 +58,22 @@ def initialize(model, options = {}, *expand)
           }
         end
 
+        link :lock do
+          {
+            href: api_v3_paths.user_lock(represented.id),
+            title: "Set lock on #{represented.login}",
+            method: :post
+          } if current_user_is_admin && represented.status == User::STATUSES[:active]
+        end
+
+        link :unlock do
+          {
+            href: api_v3_paths.user_lock(represented.id),
+            title: "Remove lock on #{represented.login}",
+            method: :delete
+          } if current_user_is_admin && represented.status == User::STATUSES[:locked]
```

Comment:
How about adding a ```lockable?(user)``` method to the user/principal/group model to handle this on a more appropriate level. The group could then always respond with false, the user could respond based on the logic shown here.

Same should apply for ```activatable?(user)```.

I guess not only the API would benefit from this.